docs(AwsRedisInstance): add resources and tutorial (#619)
kyma-project/cloud-manager: docs/user/tutorials/01-70-aws-redis-instance.md

# AWS Redis Instance Tutorial
This tutorial shows how to instantiate Redis and connect to it.
s
## Simple Example

This example showcases how to instantiate Redis, connect a Pod to it, and send a PING command.

1. Instantiate Redis

> 

Author: Alright, I will add the lines when the review is finished, so I don't make the rows with comments outdated.

> [!NOTE]
> This action may take 10+ min

```yaml
apiVersion: cloud-resources.kyma-project.io/v1beta1
kind: AwsRedisInstance
metadata:
  name: awsredisinstance-simple-example
spec:
  cacheNodeType: cache.t3.micro
```

2. Instantiate the redis-cli Pod

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: awsredisinstance-simple-example-probe
spec:
  containers:
  - name: redis-cli
    image: redis:latest
    command: ["/bin/sleep"]
    args: ["999999999999"]
    env:
    - name: HOST
      valueFrom:
        secretKeyRef:
          key: host
          name: simple-example-probe
    - name: PORT
      valueFrom:
        secretKeyRef:
          key: port
          name: simple-example-probe
```

> 3. Exec into the Pod:

Author: It is actually a command called `kubectl exec`. Should this still be changed?

> 3. Exec into the Pod:

Reviewer: I see! Then don't worry; keep "exec".

```bash
kubectl exec -i -t awsredisinstance-simple-example-probe -c redis-cli -- sh -c "clear; (bash || ash || sh)"
```

4. Exec a PING command:

```bash
redis-cli -h $HOST -p $PORT
```
You should receive `PONG` back from the server.

## Complex Example

> This example showcases how to instantiate Redis by using most of the spec fields, connect a Pod to it, and send a PING command.

Reviewer: That makes a lot of sense! 
Then I'd suggest something like this - making the headlines more meaningful along the way.

> ## Basic Setup
> If you don't need auth or TLS, you can instantiate Redis and connect the Pod with only the required fields:
> (...)

> ## Setup with Auth and TLS
> If you want to instantiate Redis and connect the Pod with auth and TLS, use the following setup:
> (...)

> This example showcases how to instantiate Redis by using most of the spec fields, connect a Pod to it, and send a PING command.

Author: Hmm, that way it might sound like examples are only different due to auth and TLS, but in fact, the complex example is also showcasing how to specify Redis version, configuration, and maintenance policy.

> This example showcases how to instantiate Redis by using most of the spec fields, connect a Pod to it, and send a PING command.

Reviewer: Oh, I see! How about:

> ## Minimal Setup
> To instantiate Redis and connect the Pod with only the required fields, use the following setup:

> ## Advanced Setup
> To specify advanced features (such as Redis version, configuration, and maintenance policy) and set up auth and TLS, use the following setup:

> This example showcases how to instantiate Redis by using most of the spec fields, connect a Pod to it, and send a PING command.

Author: Perfect! I will use that!

1. Instantiate Redis

> [!NOTE]
> This action may take 10+ min
```yaml
apiVersion: cloud-resources.kyma-project.io/v1beta1
kind: AwsRedisInstance
metadata:
  name: awsredisinstance-complex-example
spec:
  cacheNodeType: cache.t3.micro
  engineVersion: "7.0"
  authEnabled: true
  transitEncryptionEnabled: true
  parameters:
    maxmemory-policy: volatile-lru
    activedefrag: "yes"
  preferredMaintenanceWindow: sun:23:00-mon:01:30
  autoMinorVersionUpgrade: true
```

2. Instantiate the redis-cli Pod

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: awsredisinstance-complex-example-probe
spec:
  containers:
  - name: redis-cli
    image: redis:latest
    command: ["/bin/sleep"]
    args: ["999999999999"]
    env:
    - name: HOST
      valueFrom:
        secretKeyRef:
          key: host
          name: awsredisinstance-complex-example
    - name: PORT
      valueFrom:
        secretKeyRef:
          key: port
          name: awsredisinstance-complex-example
    - name: AUTH_STRING
      valueFrom:
        secretKeyRef:
          key: authString
          name: awsredisinstance-complex-example
```

> 3. Exec into the Pod:

Reviewer: ignore :)

```bash
kubectl exec -i -t awsredisinstance-complex-example-probe -c redis-cli -- sh -c "clear; (bash || ash || sh)"
```

> 4. Install and update ca-certificates:

Reviewer: oh this is different from the simple example, too!

```suggestion
4. Install and update ca-certificates:
```

```bash
apt-get update && \
  apt-get install -y ca-certificates && \
  update-ca-certificate
```

> 5. Exec a PING command:

Reviewer: ```suggestion
5. Run a PING command:
```

```bash
redis-cli -h $HOST -p $PORT -a $AUTH_STRING --tls
```

> You should receive `PONG` back from the server.

Reviewer: ```suggestion
If your setup was successful, you get `PONG` back from the server.
```
